### PR TITLE
Pubsub Subscribe/Unsubscribe ordering Bug Fix

### PR DESCRIPTION
**Pull request by howawong**
Connects skygeario/chat#155

### CODE REVIEW

File: Swift Chat Demo 2/Info.plist [hunk 1]
```diff
@@ -46,6 +46,11 @@
 	<key>NSPhotoLibraryAddUsageDescription</key>
 	<string>The app requires access the photo library for sending image message</string>
 	<key>NSMicrophoneUsageDescription</key>
-	<string>The app requires access the microphone for sending voice message</string>
+    <string>The app requires access the microphone for sending voice message</string>
+    <key>NSAppTransportSecurity</key>
+    <dict>
+        <key>NSAllowsArbitraryLoads</key>
+        <true/>
+    </dict>
```
Comment (carmenlau) [inline Swift Chat Demo 2/Info.plist]: Please remove `NSAppTransportSecurity` setting. We should not add this to the PR.

File: Swift Chat Demo 2/ConversationListViewController.swift [hunk 2]
```diff
@@ -48,18 +48,16 @@ class ConversationListViewController: UIViewController {
...
             SVProgressHUD.showError(withStatus: error.localizedDescription)
         }
-    }
-    
-    override func viewDidAppear(_ animated: Bool) {
         subscribeUserChannel()
     }
     
-    override func viewDidDisappear(_ animated: Bool) {
+    override func viewWillDisappear(_ animated: Bool) {
```
Comment (carmenlau) [inline Swift Chat Demo 2/ConversationListViewController.swift:60]: Would you mind help adding `super` to functions like `viewWillDisappear`, `viewWillAppear`... etc? From the doc, we must call super for those functions.

https://developer.apple.com/documentation/uikit/uiviewcontroller/1621485-viewwilldisappear

**Review by carmenlau — CHANGES_REQUESTED** [review]

**Comment by carmenlau** [comment]
@howawong Could you help rebase? thanks!

**Comment by howawong** [comment]
Continue at #25

**Pull request closed without merging**